components/portal/messages/controllers.js

```diff
@@ -205,13 +205,19 @@ define(['angular'], function(angular) {
             // Else hide messages features
             if (angular.equals($scope.$parent.showMessagesFeatures, true)) {
               configureNotificationsScope();
+              checkForChangesInOtherInstances;
             } else {
               vm.showMessagesFeatures = false;
               vm.isLoading = false;
             }
           }
         });
 
+        var checkForChangesInOtherInstances = $rootScope.$watch(
```

Comment:
I'd like to suggestusing a service and `$broadcast` that we have in place already instead of adding more variability to the notifications logic that I already made very complicated.

Take a look at the [messagesService](https://github.com/uPortal-Project/uportal-app-framework/blob/master/components/portal/messages/services.js#L337). We can leverage that to add a number (count) to what we're broadcasting so that the same listener we're using in the [main menu](https://github.com/uPortal-Project/uportal-app-framework/blob/master/components/portal/main/controllers.js#L206) can be reused in the priority notifications.

Since you just left for vacation, I would be willing to submit a separate PR with that model if you want.

Reply:
The problem is that there are two instances of that controller, and they don't know about each other - even when the common service is invoked.

Reply:
Do the controllers need to know about each other? Some kind of pub-sub pattern might work so that each controller only needs to know about the (singleton) service, the controllers call the service to make changes, and the service publishes the updated state to subscribed controllers?

This is one of those moments when the best way to talk about code is with code. That separate PR from @thevoiceofzeke would be interesting and we'd learn from it whether we ended up adopting it or not.